### ISSUE TITLE
[All the mods 3 V.2.8] Storage scanner not consumeing items when crafting with it

### ISSUE THREAD

**Issue by Umber0010** [McJty/RFTools#1334]
the title says most of it, the storage scanner I have connected to my storage system is not using items when I craft something (ie, making a crafting table will give me the table but not use planks). I'm certain the reason this is happening is because i'm using an AA item laser relay system for organization (i'm using crates for the storage itself)  I believe the issue is caused because I have the interface for said laser relay system high in the inventory list. the reason I see this as an issue is because the scanner can see all items  connected to the interface, but the interface can not be extracted from, only inserted into. so while I don't know exactly how crafting works with the scanner. this is likely caused because the crafting grid isn't an inventory in itself. 

And just because that was worded weird (sorry about that, I didn't realize the possible issue with the scanner itself until after I typed most of that)

TL;DR: the storage scanner is not using items when it crafts stuff, I believe this is caused because of a combination of the item interface from AA seeing  all items in the laser network, but not being able to be extracted from, and possible the Storage scanner not having an inventory for the crafting grid (again I have no idea how the scanner works behind the scenes)

**Comment by McJty** [comment]
What version of RFTools is this (I don't know all the modpacks)?

I have not used AA yet so can you explain to me what that laser relay system does?

Where are the actual items stored?

**Comment by Umber0010** [comment]
let's see, the file is called rftools-1.12-6.12.jar

the laser item relay system is, to put it simply, immersive engineering wires, but for items. here, this should explain it better than I can. http://ellpeck.de/actaddmanual/#itemRelays

the are being storen in Acctually Additions storage crates, which are simply just really, really, really big chests http://ellpeck.de/actaddmanual/#crate

**Comment by McJty** [comment]
How is that relay system used in conjunction with the storage scanner? I mean the storage scanner can directly access the crates. What is happening in your case?

**Comment by Umber0010** [comment]
i'm only letting the scanner route items to an input chest which sends items to an interface to be sorted by mod, the interface is high in the list so it's the first inventory it sees, however, it can not extract items from the interface.

**Comment by McJty** [comment]
Not exactly following. Can you describe this setup in detail so that I can replicate it?

**Comment by Umber0010** [comment]
Np, I'll do it tomorrow once I've had some sleep

**Comment by Umber0010** [comment]
Alright, it starts with an input crate which has items piped into an item interface. This interface has an item laser relay on top. This is linked to a storage drawer system on the right side of the room for metals and other things I have a lot of. Then, on the left side of the room I have 10 crates with lasers attached, 1 of them is an overflow crate, and the other 9 are set to filter by mod. Then in my scanner I only have the interface to routable. I can confirm it is an issue with the interface as when I did we routing from the interface and enable it on the drawer controller it does use items in crafting. I'll send picks after this play session

**Comment by McJty** [comment]
'This is linked to a storage drawer system' How would that work? You mean you select the interface from the scanner?

**Comment by Umber0010** [comment]
for the drawer system, I just put a relay onto a drawer controller, and yes, that is what I mean
![2017-08-14_07 06 46](https://user-images.githubusercontent.com/25757398/29271369-117dc01e-80c9-11e7-8356-91f9bd9d9518.png)
![2017-08-14_07 16 58](https://user-images.githubusercontent.com/25757398/29271374-18907f0e-80c9-11e7-92f1-2ed857c57f8c.png)
aright, let's see if that works

**Comment by McJty** [comment]
I suspect this is a bug in AA as I'm using regular ItemHandler methods to get the items

**Comment by Umber0010** [comment]
hmmmm, maybe. i'm still sure it's because the interface can't export though, i'll go to his GitHub and try to work it out
